crates/ruff_linter/src/rules/flake8_simplify/rules/yoda_conditions.rs

```diff
@@ -78,18 +80,51 @@ impl Violation for YodaConditions {
     }
 }
 
-/// Return `true` if an [`Expr`] is a constant or a constant-like name.
-fn is_constant_like(expr: &Expr) -> bool {
+/// Comparisons left hand side must not be more [`ConstantLikelihood`] than their right hand side
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+enum ConstantLikelihood {
+    Unlikely = 0,
+    Probably = 1,   // CAMEL_CASED vars
+    Definitely = 2, // literals, empty dicts and tuples...
+}
+
+fn how_likely_constant_str(s: &str) -> ConstantLikelihood {
+    if str::is_cased_uppercase(s) {
+        ConstantLikelihood::Probably
+    } else {
+        ConstantLikelihood::Unlikely
+    }
+}
+
+/// Return [`Expr`] [`ConstantLikelihood`] level depending on simple heuristics.
+fn how_likely_constant(expr: &Expr) -> ConstantLikelihood {
```

Comment:
I'd probably add an associated method on `ConstantLikelihood`, like:

```rust
impl ConstantLikelihood {
  fn from_expression(expr: &Expr) -> Self { ... }
}
```

It helps with discoverability, since this is kind of a constructor.

Reply:
like this? 
https://github.com/astral-sh/ruff/blob/7879f9e921c3a67f1b89c290e591bb01799b5479/crates/ruff_linter/src/rules/flake8_simplify/rules/yoda_conditions.rs#L96-L143

Reply:
Yup, exactly!

Reply:
If we didn't need `preview`, we could also do...

```rust
impl From<&Expr> for ConstantLikelihood {
  ...
}
```

That would enable us to do things like `expr.into()`.